server-1927| moving code from internal repo to external (#14)
CircleCI-Public/server-scripts: kots-exporter/README.md

# server-kots-exporter [Beta]

> This is a simple script to download KOTS config and convert it in `helm` value file

Reviewer: ```suggestion
This is a script to download KOTS config and convert it to a `helm` values file
```

## Prerequisite
- [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
- [yq](https://github.com/mikefarah/yq#install)
- [helm](https://github.com/helm/helm#install) (v3.x)
- [helm-diff](https://github.com/databus23/helm-diff#install) (Good to have for comparing helm releases)
- [velero](https://velero.io/docs/v1.6/contributions/minio/#back-up) (Good to have for k8s backup/restore)

### Docker secret `regcred` must be exists
The Helm chart references many private, proprietary container images. The DockerHub
API token you've been supplied will allow you to pull these images. To do so we need
to create a Docker registry secret in your Kubernetes cluster:

```
$ kubectl create secret docker-registry regcred \

>   --docker-server=https://index.docker.io/v1/ \

Reviewer: Not sure if this is the right registry address and path, but it will be in azure's `cciserver`
```suggestion
  --docker-server=https://cciserver.azurecr.io/ \
```

  --docker-username=<image-registry-username> \
  --docker-password=<image-registry-password> \
  --docker-email=<notification-email-id>
```

> docker token can be generated [here](https://hub.docker.com/settings/security) for your account to use as `image-registry-password`.

Reviewer: We are giving them the username and password
```suggestion

```

## Reason

> To enable user to download the existing `helm` values file from kots installation which can be utilized with `helm install/upgrade` commands

Reviewer: I am not sure what this section is referring too
